requre/helpers/tempfile.py

```diff
@@ -24,30 +24,53 @@
 import os
 import logging
 from typing import Optional
-from requre.storage import StorageCounter
+from requre.cassette import Cassette
+from requre.objects import ObjectStorage
 
 logger = logging.getLogger(__name__)
 
 
-class TempFile(StorageCounter):
+class TempFile(ObjectStorage):
     """
     replace system tempfile module with own predictable names implementation
      of temp files for mocking
     """
 
     root = "/tmp"
     prefix = "static_tmp"
+    _cassette: Cassette = None
+    _cassette_file = None
+    counter = 0
+    dir_suffix = "file_storage"
+
+    @classmethod
+    def set_cassette(cls, value: Cassette):
```

Comment:
This is only a temporary solution for backwards compatibility, isn't it?

Can we have a comment and/or deprecation warning?

Reply:
this is not for backward compatibility I think :-)

Reply:
So I would try not to use class attributes if possible...;)

Reply:
yep I understand, but it needs next round of refactoring :-)

This change now changes behaviour of files and tempfile to inherit behaviour from Object what uses it.

Reply:
Is it too hard to make it properly? I can take a look on Friday if you want.

If it blocks the OGR test-rewrite, we can skip some tests or just don't merge any change that influence the requre responses (not a big problem).


To be honest, I don't really understand the PR and need to spend some more time to make a good judgement.